### PR TITLE
Fix various things with reads/search

### PR DESCRIPTION
**Pull request by dcolligan**
Issue #492
Issue #496

### CODE REVIEW

File: ga4gh/datamodel/reads.py [hunk 1]
```diff
@@ -232,15 +233,15 @@ def getReadAlignments(self, referenceId=None, start=None, end=None):
...
-            referenceName = samFile.getrname(referenceId)
-        referenceName, start, end = self.sanitizeAlignmentFileFetch(
-            referenceName, start, end)
+        referenceId, start, end = self.sanitizeAlignmentFileFetch(
+            referenceId, start, end)
         # TODO deal with errors from htslib
-        readAlignments = samFile.fetch(referenceName, start, end)
+        try:
+            readAlignments = samFile.fetch(referenceId, start, end)
+        except ValueError:
```
Comment (jeromekelleher) [inline ga4gh/datamodel/reads.py]: Hmmm --- is ReferenceNotFound the _only_ reason we could ever get a ValueError here? I think all errors coming up from Pysam are ValueErrors, so this doesn't really distinguish. Isn't there some way we can test if the referenceId exists before we run the query? This way we know what the problem is and can raise the correct exception.

File: tests/end_to_end/test_remote_fetch.py [hunk 2]
```diff
@@ -42,6 +45,11 @@ def _assertLogsWritten(self):
         # client should have gotten some reads back
         self.assertEqual(len(self.client.getOutLines()), 10)
 
+    def tearDown(self):
+        # clean up downloaded local index file
+        for indexFile in glob.glob("*.bai"):
```
Comment (jeromekelleher) [inline tests/end_to_end/test_remote_fetch.py]: Where do these get written? Shouldn't this be managed by the `cleanupHtslibMess` in datamodel?

File: tests/unit/test_client.py [hunk 3]
```diff
@@ -167,12 +166,33 @@ def testRunSearchRequest(self):
...
             # verify requests.post called correctly
+            httpMethod = 'POST'
             url = "http://example.com/referencesets/search"
-            jsonString = protocolRequest.toJsonString()
+            data = protocolRequest.toJsonString()
             headers = {"Content-type": "application/json"}
-            httpMethod = 'POST'
-            mockPost.assert_called_twice_with(
-                httpMethod, url, jsonString, headers=headers, verify=False)
+            params = {u'key': u'KEY'}
```
Comment (jeromekelleher) [inline tests/unit/test_client.py]: This new block and the next one look almost identical. Would it be worth making a helper method to do this checking?

**Comment by jeromekelleher** [comment]
Looks great, thanks @dcolligan. I think the error thing is important, but the rest of the comments are just minor nits.

File: tests/end_to_end/test_remote_fetch.py (hunk 2 above)
Comment (dcolligan) [inline tests/end_to_end/test_remote_fetch.py]: they get written in the current directory.  let's put this in as a stop-gap and then file another issue to take care of this for real.

File: ga4gh/datamodel/reads.py (hunk 1 above)
Comment (dcolligan) [inline ga4gh/datamodel/reads.py]: yeah, good point.  the remedy will involve moving some pysam-like logic up into our code, but I guess the alternative (catching a generic ValueError) is worse.

File: tests/end_to_end/test_remote_fetch.py (hunk 2 above)
Comment (dcolligan) [inline tests/end_to_end/test_remote_fetch.py]: I've filed #500 ... I'll take these out in the mean time

File: tests/unit/test_client.py (hunk 3 above)
Comment (dcolligan) [inline tests/unit/test_client.py]: Sure.  FYI, these lines needed to be changed after I did a library update.  Something with the interaction of requests / mock / something else?

**Comment by dcolligan** [comment]
@jeromekelleher pushed an updated commit

File: tests/unit/test_client.py (hunk 3 above)
Comment (jeromekelleher) [inline tests/unit/test_client.py]: Ah right, now I see. Maybe we should start pinning versions by default for libs --- this sort of thing is very disruptive.

**Comment by jeromekelleher** [comment]
Excellent stuff, thanks @dcolligan. +1

**Comment by dcolligan** [comment]
@jeromekelleher cool, why don't we merge this now since develop is broken without it (due to the library version issue)

**Pull request merged**